ark/type/test/array.test.ts

```diff
@@ -32,7 +32,22 @@ suite("array", () => {
 			attest(t.infer).typed as string[]
 			attest(t.condition).equals(type("string[]").condition)
 		})
-
+		suite("optional tuple literals", () => {
```

Comment:
These should be split up into more individual cases.  E.g. the duplicate ? is its own case because it has its own logic. Each type of error would be its own case.

I assume all of this is borked ATM because I screwed up the repo, but it should still be split up into the cases we want to test